examples/timeseries/timeseries_traffic_forecasting.py

```diff
@@ -0,0 +1,655 @@
+"""
+To reduce the problem size and make the training faster, we will only
+work with a sample of 26 roads out of the 228 roads in the dataset.
+We have chosen the roads by starting from road 0, choosing the 5 closest
+roads to it, and continuing this process until we get 25 roads.
+`sample_routes` contains the IDs of the selected roads.
+"""
+
+sample_routes = [
+    0,
+    1,
+    4,
+    7,
+    8,
+    11,
+    15,
+    108,
+    109,
+    114,
+    115,
+    118,
+    120,
+    123,
+    124,
+    126,
+    127,
+    129,
+    130,
+    132,
+    133,
+    136,
+    139,
+    144,
+    147,
+    216,
+]
+route_distances = route_distances[np.ix_(sample_routes, sample_routes)]
+speeds_array = speeds_array[:, sample_routes]
+
+print(f"route_distances shape={route_distances.shape}")
+print(f"speeds_array shape={speeds_array.shape}")
+
+"""
+### Data visualization
+"""
+
+"""
+Here are the timeseries of the traffic speed for two of the routes:
+"""
+
+plt.figure(figsize=(18, 6))
+plt.plot(speeds_array[:, [0, -1]])
+plt.legend(["route_0", "route_25"])
+
+"""
+We can also visualize the correlation between the timeseries in different routes.
+"""
+
+plt.figure(figsize=(8, 8))
+plt.matshow(np.corrcoef(speeds_array.T), 0)
+plt.xlabel("road number")
+plt.ylabel("road number")
+
+"""
+Using this correlation heatmap, we can see that for example the speed in
+routes 4, 5, 6 are highly correlated.
+"""
+
+"""
+### Splitting and normalizing data
+"""
+
+"""
+Next, we split the speed values array into train/validation/test sets
+and normalize the resulting arrays:
+"""
+
+train_size, val_size = 0.5, 0.2
+
+
+def preprocess(data_array: np.ndarray, train_size: float, val_size: float):
+    """Preprocesses data.
+    This function splits data into train/val/test sets and normalizes the data.
+
+    :param data_array: ndarray of shape [n_time_steps, n_routes]
+    :param train_size:
+    :param val_size:
+    :return:
+    """
+
+    n_time_steps = data_array.shape[0]
+    n_train, n_val = int(n_time_steps * train_size), int(n_time_steps * val_size)
+    train_array = data_array[:n_train]
+    mean, std = train_array.mean(axis=0), train_array.std(axis=0)
+
+    train_array = (train_array - mean) / std
+    val_array = (data_array[n_train : (n_train + n_val)] - mean) / std
+    test_array = (data_array[(n_train + n_val) :] - mean) / std
+
+    return train_array, val_array, test_array
+
+
+train_array, val_array, test_array = preprocess(speeds_array, train_size, val_size)
+
+print(f"train set size: {train_array.shape}")
+print(f"validation set size: {val_array.shape}")
+print(f"test set size: {test_array.shape}")
+
+"""
+### Creating Tensorflow datasets
+"""
+
+"""
+Next, we create the datasets for our forecasting problem. Let $v^i_t$
+be the speed on road $i$ at time $t$. Then the vector $V_t=[v^1_t,...,v^N_t]$
+will contain the speed for all $N$ roads at time $t$. The forecasting problem
+can be stated as follows: given a sequence $V_{t+1},...,V_{t+T}$ of the
```

Comment:
No latex notation. Please make sure to review our contributions guidelines: https://github.com/keras-team/keras-io/blob/master/contributor_guide.md